zkabacus-crypto/src/proofs.rs

```diff
@@ -810,4 +812,73 @@ mod tests {
             .verify(&merchant_params, &public_values, &context)
             .is_some());
     }
+
+    #[test]
+    #[cfg(feature = "bincode")]
+    fn test_pay_proof_with_serialize() {
+        let merchant_balance = 0;
+        let customer_balance = 100;
+        let amount = 10;
+        let pay = PaymentAmount::pay_merchant;
+        let mut rng = rng();
+        let merchant_params = merchant::Config::new(&mut rng);
+        let params = merchant_params.to_customer_config();
+
+        // Create a state.
+        let channel_id = channel_id(&mut rng);
+        let old_state = State::new(
+            &mut rng,
+            channel_id,
+            MerchantBalance::try_new(merchant_balance).unwrap(),
+            CustomerBalance::try_new(customer_balance).unwrap(),
+        );
+
+        // Update state.
+        let amount = pay(amount).unwrap();
+        let new_state = old_state.apply_payment(&mut rng, amount).unwrap();
+
+        // Run establish proof and sign result to get a valid signature on the old state.
+        let (verified_state_com, old_pt_bf) =
+            run_establish_proof(merchant_balance, customer_balance);
+        let pay_token = BlindedPayToken::sign(&mut rng, &merchant_params, verified_state_com)
+            .unblind(old_pt_bf);
+
+        // Save a copy of the nonce...
+        let nonce = *old_state.nonce();
+
+        let context = Context::new(b"pay proof verify test");
+
+        // Form proof.
+        let param_ser = bincode::serialize(&params).unwrap();
+        let start = Instant::now();
```

Comment:
Nonblocking: You might consider giving more meaningful names to all the `start` timers. Maybe `config_deserialize_timer` here, `proof_creation_timer` on L855, etc.